Library/QuickPulseSender.ts

```diff
@@ -47,12 +49,20 @@ class QuickPulseSender {
 
         const req = https.request(options, (res: http.IncomingMessage) => {
             const shouldPOSTData = res.headers[QuickPulseConfig.subscribed] === "true";
+            this._consecutiveErrors = 0;
             done(shouldPOSTData, res);
         });
         req.on("error", (error: Error) => {
             // Unable to contact qps endpoint.
             // Do nothing for now.
-            Logging.warn("Unable to contact qps endpoint, dropping this Live Metrics packet", error);
+            this._consecutiveErrors++;
+            if (postOrPing === "post") {
+                if (this._consecutiveErrors > 20) {
+                    Logging.warn("Unable to contact qps endpoint, dropping this Live Metrics packet", error);
```

Comment:
You should follow the format used for ingestion and state how many consecutive errors have occurred. Also highly suggest using modulo instead of every request over > 20 failures for consistency.

Reply:
Also, calls to `Logging` should get passed a tag to indicate where the failure came from

Reply:
Updated